core/go/internal/sequencer/coordinator/snapshot.go

```diff
@@ -0,0 +1,130 @@
+	"github.com/LFDT-Paladin/paladin/core/internal/sequencer/coordinator/transaction"
+	"github.com/LFDT-Paladin/paladin/sdk/go/pkg/pldtypes"
+)
+
+func action_SendHeartbeat(ctx context.Context, c *coordinator) error {
+	return c.sendHeartbeat(ctx, c.contractAddress)
+}
+
+func (c *coordinator) sendHeartbeat(ctx context.Context, contractAddress *pldtypes.EthAddress) error {
+	snapshot := c.getSnapshot(ctx)
+	log.L(ctx).Debugf("sending heartbeats for sequencer %s", contractAddress.String())
+	var err error
+	c.originatorNodePoolMutex.RLock()
+	defer c.originatorNodePoolMutex.RUnlock()
+	for _, node := range c.originatorNodePool {
+		if node != c.nodeName {
+			log.L(ctx).Debugf("sending heartbeat to %s", node)
+			err = c.transportWriter.SendHeartbeat(ctx, node, contractAddress, snapshot)
+			if err != nil {
+				log.L(ctx).Errorf("error sending heartbeat to %s: %v", node, err)
+			}
+		}
+	}
+	return err
+}
+
+func (c *coordinator) getSnapshot(ctx context.Context) *common.CoordinatorSnapshot {
+	log.L(ctx).Infof("creating snapshot for sequencer %s", c.contractAddress.String())
+	// This function is called from the sequencer loop so is safe to read internal state
+	pooledTransactions := make([]*common.Transaction, 0, len(c.transactionsByID))
+	dispatchedTransactions := make([]*common.DispatchedTransaction, 0, len(c.transactionsByID))
+	confirmedTransactions := make([]*common.ConfirmedTransaction, 0, len(c.transactionsByID))
+
+	//Snapshot contains a coarse grained view of transactions state.
+	// All known transactions fall into one of 3 categories
+	// 1. Pooled transactions - these are transactions that have been delegated but not yet dispatched
+	// 2. Dispatched transactions - these are transactions that are past the point of no return, the precise status (ready for collection, dispatched, nonce assigned, submitted to a blockchain node) is dependant on parallel processing from this point onward
+	// 3. Confirmed transactions - these are transactions that have been confirmed by the network
+	for _, txn := range c.transactionsByID {
+		log.L(ctx).Infof("next transaction to assess current status of %s. Current state: %s", txn.ID.String(), txn.GetCurrentState().String())
+		switch txn.GetCurrentState() {
+		// pooled transactions are those that have been delegated but not yet dispatched, this includes the various states from being delegated up to being ready for dispatch
+		case transaction.State_Reverted:
+			//NOOP - this transaction is just waiting to be cleaned up so we don't include it in the snapshot
+		case transaction.State_Blocked:
+			fallthrough
+		case transaction.State_Confirming_Dispatchable:
+			fallthrough
+		case transaction.State_Endorsement_Gathering:
+			fallthrough
+		case transaction.State_PreAssembly_Blocked:
+			fallthrough
+		case transaction.State_Assembling:
+			fallthrough
+		case transaction.State_Pooled:
+			pooledTransactions = append(pooledTransactions, &common.Transaction{
+				ID: txn.ID,
+			})
```

Comment:
I'm pleased (per earlier comments) it's just the transaction ID that's managed here, but `common.Transaction` seems like a wrong description of the object.

Both the `common` package feels very odd - which maybe comes to earlier comments that this really isn't part of `core` and is thinking it's big enough to have it's own package structure.

But also the `Transaction` is odd, if it's just a `TransactionReference`, or a `SnapshotPooledTransaction` (I guess with the ability for extensibility in the future - so it's not just an array of IDs.)